src/polymer-ide.ts

```diff
@@ -27,68 +27,96 @@ import TooltipManager from './tooltip-manager';
 
 interface ViewState {}
 
+type WorkSpace = {
+  linter: Linter;
+  autocompleter: Autocompleter;
+  editorService: RemoteEditorService;
+  tooltipManager: TooltipManager;
+};
+
 class PolymerIde {
   subscriptions: CompositeDisposable = null;
-  linter: Linter = null;
-  autocompleter: Autocompleter = null;
-  editorService: RemoteEditorService;
+  workspaces: Map<string, [number, WorkSpace]> = new Map();
 
   activate(_: ViewState) {
-    // Initialize.
-    this.linter = new Linter();
-    this.autocompleter = new Autocompleter();
-
     // Events subscribed to in atom's system can be easily cleaned up with a
     // CompositeDisposable
     this.subscriptions = new CompositeDisposable();
 
     this.subscriptions.add(
-        atom.project.onDidChangePaths((projectPaths: string[]) => {
-          this.setProjectPaths(projectPaths);
-        }));
-    this.setProjectPaths(atom.project.getPaths());
-    this.subscriptions.add(new TooltipManager(this.editorService));
+      atom.workspace.observeTextEditors((editor: AtomCore.IEditor) => {
+        this.updateWorkspaceMap(editor);
+      })
+    );
   };
 
-  deactivate() {
-    this.subscriptions.dispose();
-    this.subscriptions = null;
-    this.editorService = null;
-    this.linter = null;
-    this.autocompleter = null;
+  private getProjectPathFromEditor(editor: AtomCore.IEditor) {
+    return atom.project.relativizePath(editor.getPath())[0];
   };
 
-  setProjectPaths(projectPaths: string[]) {
-    if (projectPaths.length === 0) {
-      this.linter.configurationError = 'polymer-ide only works with projects.';
-    } else if (projectPaths.length > 1) {
-      this.linter.configurationError =
-          `polymer-ide only works with projects with exactly one root ` +
-          `directory, this project has: ${JSON.stringify(projectPaths)}`;
+  private updateWorkspaceMap(editor: AtomCore.IEditor) {
+    const project = this.getProjectPathFromEditor(editor);
+    editor.onDidDestroy(() => {
+      const workspace = this.workspaces.get(project);
+      workspace[0] = workspace[0] - 1;
+    });
+    if (this.workspaces.has(project)) {
+      this.workspaces.get(project)[0] = this.workspaces.get(project)[0] + 1;
     } else {
-      const rootDir = projectPaths[0];
-      if (this.editorService) {
-        this.editorService.dispose();
+      // TODO(timvdlippe): Expose as user setting?
+      if (this.workspaces.size == 10) {
```

Comment:
Should probably be `>= 10` to guard against adding a workspace if not one was removed, but also one was added.